nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/app.component.ts

```diff
@@ -37,5 +43,22 @@ export class AppComponent {
                 this.guardLoading = false;
             }
         });
+
+        let theme = this.storage.getItem('theme');
+
+        // Initially check if dark mode is enabled on system
+        const darkModeOn = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
+
+        // If dark mode is enabled then directly switch to the dark-theme
+        this.themingService.toggleTheme(darkModeOn, theme);
+
+        if (window.matchMedia) {
+            // Watch for changes of the preference
+            window.matchMedia('(prefers-color-scheme: dark)').addListener((e) => {
+                theme = this.storage.getItem('theme');
+                const newColorScheme = e.matches ? 'dark' : 'light';
```

Comment:
If `e.matches` is a boolean, do we need to convert it into a string here? Can `e.matches` be passed to `toggleTheme` just like it is above?